blaze-server/src/main/scala/org/http4s/server/blaze/Http2NodeStage.scala

```diff
@@ -185,25 +185,27 @@ private class Http2NodeStage[F[_]](
       val hs = HHeaders(headers.result())
       val req = Request(method, path, HttpVersion.`HTTP/2.0`, hs, body, attributes)
 
-      async.unsafeRunAsync {
-        try service(req)
-          .recoverWith(serviceErrorHandler(req).andThen(_.widen[MaybeResponse[F]]))
-          .handleError { _ =>
-            Response[F](InternalServerError, req.httpVersion)
-          }
-          .map(renderResponse(_))
-        catch serviceErrorHandler(req).andThen(_.widen[MaybeResponse[F]])
-      } {
-        case Right(_) =>
-          IO.unit
-        case Left(t) =>
-          IO(logger.error(t)("Error rendering response"))
-      }
+      executionContext.execute(new Runnable {
```

Comment:
This bit mirrors how Http1 works and should do the same as #1435 did for 0.17.

Reply:
cant that be done via `F.shift(executionContext) >> currentRunMethodMinusUnsafeRun`

Reply:
I thought the same, and no, because the routing happens outside the task.

```scala
scala> def foo(i: Int): IO[List[String]] = { val x = Thread.currentThread.getName; IO(List(x, Thread.currentThread.getName)) }
e; IO(List(x, Thread.currentThread.getName)) }Thread.currentThread.getNam
<console>:27: warning: parameter value i in method foo is never used
       def foo(i: Int): IO[List[String]] = { val x = Thread.currentThread.getName; IO(List(x, Thread.currentThread.getName)) }
               ^
foo: (i: Int)cats.effect.IO[List[String]]

scala> (Effect[IO].shift(global) >> foo(42)).unsafeRunSync
(Effect[IO].shift(global) >> foo(42)).unsafeRunSync
res7: List[String] = List(run-main-0, scala-execution-context-global-125)
```

I think it could be done with an `F.shift` if the `service` were called inside `F` and then flattened.